test/e2e/e2e_test.go

```diff
@@ -200,6 +202,90 @@ func TestE2EFlow(t *testing.T) {
 		VerifyIncreaseOfUserAccountCount(t, originalToolchainStatus, currentToolchainStatus, targetedJohnMur.Spec.UserAccounts[0].TargetCluster, 1)
 	})
 
+	t.Run("verify userAccount is not deleted if namespace is not deleted", func(t *testing.T) {
+		// given
+		johnMUR, err := hostAwait.WaitForMasterUserRecord(johnsmithName)
+		require.NoError(t, err)
+
+		fmt.Printf("\n >>>>>>>>>>>>>>>>>>>>>>>>>>>> MUR fetched>>>>>>>>>>>>>>>>")
+		// set deletion timestamp and add finalizer
+		deletionTS := metav1.Now()
+		johnMUR.DeletionTimestamp = &deletionTS
+		coputil.AddFinalizer(johnMUR, "finalizer.toolchain.dev.openshift.com")
```

Comment:
why do you want to add a finalizer on the MUR?

Reply:
its the entry condition to manageCleanup here https://github.com/codeready-toolchain/host-operator/blob/master/controllers/masteruserrecord/masteruserrecord_controller.go#L119 - it expects the MUR to have both a timestamp and finalizer before calling manageCleanup, which in-turn deletes userAccount.